src/Moq/LookupOrFallbackDefaultValueProvider.cs

```diff
@@ -122,7 +122,9 @@ protected internal sealed override object GetDefaultValue(Type type, Mock mock)
 			               : type.IsArray ? typeof(Array)
 			               : type;
 
-			return this.factories.TryGetValue(handlerKey, out Func<Type, Mock, object> factory) ? factory.Invoke(type, mock)
+			Func<Type, Mock, object> factory;
+			return this.factories.TryGetValue(handlerKey         , out factory) ? factory.Invoke(type, mock)
+			     : this.factories.TryGetValue(handlerKey.FullName, out factory) ? factory.Invoke(type, mock)
 			     : this.GetFallbackDefaultValue(type, mock);
```

Comment:
The default value providers will now do two dictionary lookups instead of one in many cases. In theory, that's a slight performance degradation, but it would be surprising if this mattered much in practice. (A dictionary lookup is likely nothing when compared to all the reflection and code emitting going on elsewhere.)